recursion/factorial.py

```diff
@@ -1,11 +1,12 @@
-def factorial(num):
-    if num == 1:
-        return 1
-    else:
-        return num * factorial(num - 1)
+num = int(input("Enter a number: "))
 
-def main():
-    print(factorial(7))
+factorial = 1
 
-if __name__ == '__main__':
-    main()
+if num < 0:
+   print("Sorry, factorial does not exist for negative numbers")
+elif num == 0:
+   print("The factorial of 0 is 1")
+else:
+   for i in range(1,num + 1):
+       factorial = factorial*i
+   print("The factorial of",num,"is",factorial)
```

Comment:
enter a new line.

Reply:
@t2013anurag This was supposed to be recursive as it is placed in recursion folder